### ISSUE TITLE
Odd number of forms in map literal not localised when in binding form

### ISSUE THREAD

**Issue by yuhan0** [borkdude/clj-kondo#642]
**version**

clj-kondo v2019.11.03

**platform**

native

**editor**

Emacs+flycheck-clj-kondo 

**problem**

When an invalid map with odd number of literals is present in a binding form, clj-kondo returns a parse error on the entire file instead of locating the line number of the error. 
Outside of a binding form this appears to work normally.

**repro**
```clojure
(ns repro)

(let [{:keys [a b] c}])
```
```bash
clj-kondo --lint repro.clj
repro.clj:0:0: error: can't parse src/repro.clj, No value supplied for key: c
linting took 4ms, errors: 1, warnings: 0
```
**expected behavior**

```
repro.clj:4:19: error: missing value for key c
```

**Comment by yuhan0** [comment]
The above example also doesn't pick up on the invalid binding vector, which should also present an error. (It's a fairly common typo I make when doing map destructuring in `let`)
```
repro.clj:4:6: error: let binding vector requires even number of forms
```

**Comment by borkdude** [comment]
Fixed with https://github.com/borkdude/clj-kondo/commit/89751c01a9f86310d50d6941557c91af2aa23331
